ConsoleApp1/Queue.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace ConsoleApp1
+{
+    class Queue<T>
+    {
+        // head
+        private Node<T> _head;
+        private Node<T> _tail;
```

Comment:
Why do you need _tail pointer ?

Reply:
I used it for faster adding to the end of Queue, so I didn't need to iterate from head to the end.